qemu/tests/balloon_boot_in_pause.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import random
+
+from avocado.core import exceptions
+
+from virttest import error_context
+from virttest import utils_misc
+from qemu.tests.balloon_check import BallooningTest
+
+
+class BallooningTestPause(BallooningTest):
+
+    """
+    Basic functions of memory ballooning test for guest booted
+    in paused status
+    """
+
+    def __init__(self, test, params, env):
+        super(BallooningTest, self).__init__(test, params, env)
+
+        self.vm = env.get_vm(params["main_vm"])
+        self.ori_mem = self.get_vm_mem(self.vm)
+        if self.get_ballooned_memory() != self.ori_mem:
+            self.balloon_memory(self.ori_mem)
+        self.old_mmem = self.ori_mem
+        self.old_gmem = None
+
+    @error_context.context_aware
+    def memory_check(self, step, ballooned_mem):
+        """
+        Check memory change status in monitor and return memory both
+        in guest and monitor
+        :param step: the check point string
+        :type step: string
+        :param ballooned_mem: ballooned memory in current step(compared with
+        last round of memory, i.e. self.old_mmem, instead of self.ori_mem)
+        :type ballooned_mem: int
+        :return: memory size get from monitor and guest
+        :rtype: tuple
+        """
+        error_context.context("Check memory status %s" % step, logging.info)
+        mmem = self.get_ballooned_memory()
+        gmem = self.get_memory_status()
+        if (abs(mmem - self.old_mmem)) != ballooned_mem or \
```

Comment:
The variable is a little confusing.  
 mmem = self.get_ballooned_memory()    it looks like the mmem is equal to ballooned_mem, which is not.